src/main/java/graphql/execution/MergedSelectionSet.java

```diff
@@ -13,10 +14,10 @@
 @PublicApi
 public class MergedSelectionSet {
 
-    private final Map<String, MergedField> subFields;
+    private final ImmutableMap<String, MergedField> subFields;
```

Comment:
Make an immutable map of the merged selection set of fields

This preserves order if the original map was a LinkedHashMap (which it is)